triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Converter.scala

```diff
@@ -380,14 +381,22 @@ final class Converter(
   }
 
   def fromACS(activeContracts: Seq[CreatedEvent]): SValue = {
+    val createMapByTemplateId = mutable.HashMap.empty[value.Identifier, BackStack[CreatedEvent]]
+    for (create <- activeContracts) {
+      createMapByTemplateId += (create.getTemplateId -> (createMapByTemplateId.getOrElse(
+        create.getTemplateId,
+        BackStack.empty,
+      ) :+ create))
+    }
```

Comment:
Unclear as to runtime complexity for `groupBy` (ad hoc time results suggested we could be inefficient here?), so removing complexity ambiguity.